lib/Db/ItemMapperV2.php

```diff
@@ -138,20 +139,47 @@ public function findAllForFeed(int $feedId): array
     /**
      * Delete items from feed that are over the max item threshold
      *
-     * TODO: Implement
-     *
      * @param int $threshold Deletion threshold
+     *
+     * @return int|null Removed items
+     *
+     * @throws \Doctrine\DBAL\Exception
      */
-    public function deleteOverThreshold(int $threshold)
+    public function deleteOverThreshold(int $threshold): ?int
     {
-        $builder = $this->db->getQueryBuilder();
+        $feedQb = $this->db->getQueryBuilder();
+        $feedQb->addSelect('feed_id', $feedQb->func()->count('*', 'itemCount'))
+               ->from($this->tableName)
+               ->groupBy('feed_id');
+
+        $feeds = $this->db->executeQuery($feedQb->getSQL())
+                          ->fetchAll(FetchMode::ASSOCIATIVE);
+
+        if ($feeds === []) {
+            return null;
+        }
+
+        $deleteSubQb = $this->db->getQueryBuilder();
+        $deleteSubQb->select('id')
+            ->from($this->tableName)
+            ->where('feed_id = :feedId')
+            ->andWhere('starred = 0')
+            ->setFirstResult($threshold);
```

Comment:
That is how it works now, but I don't think that makes sense. I'd like to take this opportunity to slightly simplify the query and remove unread items (since this is quite a common request)

Reply:
But shouldn't it skip/ignore higher ids (more recent ones) or order it descending? Or am I mixing something up.
I would prefer if the cleanup of unread items was configurable, considering how many unread ones I have ._.